providers/vsphere/resources/discovery.go

```diff
@@ -124,6 +124,22 @@ func discoverDatacenter(conn *connection.VsphereConnection, datacenterResource *
 			platformID := connection.VsphereResourceID(instanceUuid, mqlHost.Moid.Data)
 			clonedConfig := conn.Conf.Clone(inventory.WithoutDiscovery())
 			clonedConfig.PlatformId = platformID
+
+			labels := map[string]string{
+				"vsphere.vmware.com/name":          mqlHost.Name.Data,
+				"vsphere.vmware.com/moid":          mqlHost.Moid.Data,
+				"vsphere.vmware.com/inventorypath": mqlHost.InventoryPath.Data,
+			}
+
+			// Add tags as labels with vsphere.vmware.com/tag/ prefix
+			if mqlHost.Tags.Error == nil {
+				for _, tag := range mqlHost.Tags.Data {
+					if tagStr, ok := tag.(string); ok {
+						labels["vsphere.vmware.com/tag/"+tagStr] = "true"
+					}
+				}
+			}
```

Comment:
This looks to me the same as on L172-L186. Can you extract it into something like `createLabels`?